src/app/community/community.component.ts

```diff
@@ -60,27 +61,36 @@ export class CommunityComponent implements OnInit, AfterViewInit {
       }, (error) => this.message = 'There was a problem getting Communities');
   }
 
-  deleteClick(community) {
-    this.deleteDialog = this.dialog.open(DialogsDeleteComponent, {
+  updateClick(community, change) {
+    this.editDialog = this.dialog.open(DialogsEditComponent, {
       data: {
-        okClick: this.deleteCommunity(community),
+        okClick: this.updateCommunity(community, change),
+        changeType: change,
         type: 'community',
         displayName: community.name
       }
     });
   }
 
-  deleteCommunity(community) {
+  updateCommunity(community, change) {
     // Return a function with community on its scope to pass to delete dialog
     return () => {
     // With object destructuring colon means different variable name assigned, i.e. 'id' rather than '_id'
       const { _id: id, _rev: rev } = community;
-      this.couchService.delete('communityregistrationrequests/' + id + '?rev=' + rev)
+      if (change !== 'delete') {
+        community['registrationRequest'] = change;
+        this.couchService.put('communityregistrationrequests/' + id + '?rev=' + rev, community)
+        .subscribe((data) => {
+          this.editDialog.close();
+        }, (error) => this.editDialog.componentInstance.message = 'There was a problem accepting this community');
+      } else {
+        this.couchService.delete('communityregistrationrequests/' + id + '?rev=' + rev)
         .subscribe((data) => {
           // It's safer to remove the item from the array based on its id than to splice based on the index
           this.communities.data = this.communities.data.filter((comm: any) => data.id !== comm._id);
           this.deleteDialog.close();
```

Comment:
You are now initializing this as `editDialog`, so this will not close the dialog.  Please make sure to try out your code before pushing it.

You should replace all references of `deleteDialog` to `editDialog` (or `promptDialog` -- see my other comment).